### DISCUSSION TITLE
Supabase Schema Visualiser doesn't open to a layout that's readable

### DISCUSSION THREAD

**Discussion by kellemar** [supabase/supabase#16989] [Contribute to Supabase]
When I try to use it to show a visual of my schema (public), everything is showed vertically stretched out. This isn't helpful for me, and it doesn't save the repositioning of my tables.

https://pbs.twimg.com/media/F3J__MjaIAAwti2?format=png&name=large

**Comment by gregnr** [comment]
Thanks for reporting @kellemar. Any chance you could share your table schemas to help us reproduce?

cc @kamilogorek in case you have any thoughts

**Comment by kellemar** [comment]
Hi @gregnr is there an email I can send to for my schemas?

**Comment by kellemar** [comment]
This is how it looks like for my tables with Strapi CMS too.

![Screenshot 2023-08-15 at 10 04 07 AM](https://github.com/supabase/supabase/assets/350990/307e8f78-42fa-4139-a0b7-719d23f57d83)

Is there an email I can share the schemas too?

**Comment by kamilogorek** [comment]
Hey @kellemar, what's happening here is that degree (and any other algorithm for calculating graphs), is ranking each node from left to right, depending on the connection nodes. If there are plenty of standalone tables, they will all land on the 0-rank, making the whole tree very tall.
There's no clean way to achieve a nice even spread of standalone nodes with Reactflow, that would fill remaining empty space.

What we can try to make it slightly better from a UX perspective, is to cap the minimum zoom that the graph will render, and add a minimap, to make sure that people won't miss the fact that there are more nodes outside of the visible area.

To change min-zoom, add `minZoom={0.8}` to https://github.com/supabase/supabase/blob/d9bf5d21f2e87e73010dc013bc70608748b7f795/studio/components/interfaces/Database/Schemas/SchemaGraph.tsx#L339-L339
To add minimap, add `<MiniMap />` with [appropriate props ](https://reactflow.dev/docs/api/plugin-components/minimap/) to https://github.com/supabase/supabase/blob/d9bf5d21f2e87e73010dc013bc70608748b7f795/studio/components/interfaces/Database/Schemas/SchemaGraph.tsx#L340-L343

**Comment by gregnr** [comment]
These are great suggestions @kamilogorek. Would love to see a PR from someone to apply these!

@kellemar ideally we share the schema publicly in this thread so that contributors can use it to test. If your schema is sensitive, then definitely no need to share it - but instead let's come up with another example schema that can be shared and referenced so that anyone can reproduce it.

**Comment by kellemar** [comment]
Hi @gregnr thanks for this. Unfortunately, I I can't share it as it belongs to the company. 

But some info; it has 64 tables in one schema, an
<img width="944" alt="Screenshot 2023-08-23 at 12 04 01 AM" src="https://github.com/supabase/supabase/assets/350990/df950436-a710-478e-ac90-ce8f009ceffd">
d on datagrip, here's how it looks like zoomed out.

**Comment by futbotism** [comment]
@gregnr I'm having the same issue, here is my schema. im wondering if this is something wrong with my schema. some edge case or breaking some foundational rules of RDB

```
create type "public"."app_role" as enum ('admin', 'owner', 'captain', 'user');

create type "public"."direction" as enum ('long', 'short');

create type "public"."user_status" as enum ('ONLINE', 'OFFLINE');

create table "public"."idea" (
    "id" uuid not null,
    "created_at" timestamp with time zone default now(),
    "title" text not null,
    "description" character varying,
    "pair" text not null,
    "direction" direction not null,
    "group_id" uuid not null,
    "created_by" uuid not null
);


alter table "public"."idea" enable row level security;

create table "public"."join_request" (
    "id" uuid not null default gen_random_uuid(),
    "created_at" timestamp with time zone default now(),
    "user_id" uuid not null,
    "group_id" uuid not null
);


alter table "public"."join_request" enable row level security;

create table "public"."group" (
    "id" uuid not null default gen_random_uuid(),
    "created_at" timestamp with time zone default now(),
    "title" text not null,
    "description" text,
    "slug" text not null
);


alter table "public"."group" enable row level security;

create table "public"."membership" (
    "id" uuid not null default gen_random_uuid(),
    "created_at" timestamp with time zone default now(),
    "user_id" uuid not null,
    "membership_type" app_role default 'user'::app_role,
    "group_id" uuid
);


alter table "public"."membership" enable row level security;

create table "public"."profile" (
    "id" uuid not null,
    "display_name" text
);
```
